gitlab-runner/src/job.rs

```diff
@@ -91,6 +105,35 @@ impl Dependency<'_> {
         self.dependency.artifacts_file.as_ref().map(|a| a.size)
     }
 
+    async fn download_impl(
+        &self,
+        writer: Arc<AsyncMutex<impl AsyncWrite + Unpin>>,
+    ) -> Result<(), ClientError> {
+        let strategy = FibonacciBackoff::from_millis(100).map(jitter).take(10);
```

Comment:
100ms seems aggressive; 10 times without jitter is:
```
=> 100ms
=> 100ms
=> 200ms
=> 300ms
=> 500ms
=> 800ms
=> 1.3s
=> 2.1s
=> 3.4s
=> 5.5s
Total: 14.3s
```

starting from 500 would be:
```
> 500ms
=> 500ms
=> 1s
=> 1.5s
=> 2.5s
=> 4s
=> 6.5s
=> 10.5s
=> 17s
=> 27.5s
Total: 71.5s
```

Ofcourse this is a bit finger in the air as there is no way to know what he remote issue is; But a less aggressive retry might probably be better